### PR TITLE
chore: add tsg link in error messages

### LINKED ISSUE #36113: [AppConfig] Add aka.ms links from SDK exception messages to anchors in the TSG
1, Create an aks link to represent the TROUBLESHOOTING(TSG) guide
2. Use the new aks link in the common Exception type, and add it to the message.
You will find an example in Identity:
https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/identity/azure-identity/src/main/java/com/azure/identity/implementation/IdentityClientBase.java#L479
Should be similar in .NET language.

### PR DESCRIPTION
**Pull request by jorgerangel-msft**
This PR enhances the error handling done by the client by appending a troubleshooting message that contains the app config [TSG link](https://aka.ms/azsdk/net/appconfiguration/troubleshoot) to error messages returned by operations within the client.

### CODE REVIEW

File: sdk/appconfiguration/Azure.Data.AppConfiguration/src/ConfigurationClient.cs [hunk 1]
```diff
@@ -1469,16 +1496,31 @@ private static RequestContext CreateRequestContext(ErrorOptions errorOptions, Ca
 
         private class ConfigurationRequestFailedDetailsParser : RequestFailedDetailsParser
         {
+            private const string TroubleshootingText =
```
Comment (jsquire) [inline sdk/appconfiguration/Azure.Data.AppConfiguration/src/ConfigurationClient.cs:1499]: This is too detailed, in my opinion.  We want to keep the focus on the actual error message rather than our TSG text.  This should be a generic message we tack onto the very end of the error message that just makes folks aware that it exists, like:

> For troubleshooting information, see https://aka.ms/azsdk/net/appconfiguration/troubleshoot

This is the form used for the other .NET libraries enriching errors.  Examples:

- [Event Hubs](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/eventhub/Azure.Messaging.EventHubs.Shared/src/Resources.Designer.cs#L918)
- [Service Bus](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/servicebus/Azure.Messaging.ServiceBus/src/Constants.cs#L57)
- [Identity](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/identity/Azure.Identity/src/Credentials/ManagedIdentityCredential.cs#L27)

File: sdk/appconfiguration/Azure.Data.AppConfiguration/tests/ConfigurationLiveTests.cs [hunk 2]
```diff
@@ -190,6 +191,8 @@ public async Task DeleteSettingWithLabel()
                 });
 
                 Assert.AreEqual(404, e.Status);
+                Assert.IsNotEmpty(e.Message);
+                Assert.IsFalse(e.Message.Contains(troubleshootingLink));
```
Comment (kinelski) [inline sdk/appconfiguration/Azure.Data.AppConfiguration/tests/ConfigurationLiveTests.cs:195]: This new assertion is not related to what this test is validating (deletion of settings with labels). I believe the mock test you created should be enough to validate this behavior. If you feel strongly about having a live test, I recommend creating a new one.

File: sdk/appconfiguration/Azure.Data.AppConfiguration/src/ConfigurationClient.cs (hunk 1 above)
Comment (kinelski) [reply sdk/appconfiguration/Azure.Data.AppConfiguration/src/ConfigurationClient.cs:1499]: I agree with Jesse here. Based on similar experiences, I believe we'll often forget to update this logic if we ever update the troubleshooting guide. This is specially true for AppConfiguration where ownership rotates frequently.

Comment (pallavit) [reply sdk/appconfiguration/Azure.Data.AppConfiguration/src/ConfigurationClient.cs:1499]: Should we have a static analyzer to check for this text field value? It may be useful in general. Also, curious if this is something we are adding to our DPG code path.

Comment (jsquire) [reply sdk/appconfiguration/Azure.Data.AppConfiguration/src/ConfigurationClient.cs:1499]: I don't think so, and no, we're not.   Troubleshooting Guides are not generated, they are created by hand.  They often don't appear until a stable release or two.   They can't be generated upfront, because we can't predict the timing.

Comment (pallavit) [reply sdk/appconfiguration/Azure.Data.AppConfiguration/src/ConfigurationClient.cs:1499]: I wasn't talking about generating the TSG in an automatable way. Just checking that we are consistent in TSGs like we are following generic text, aka.ms link in case it is present.

Comment (jsquire) [reply sdk/appconfiguration/Azure.Data.AppConfiguration/src/ConfigurationClient.cs:1499]: I was talking about generating the error message enrichment.   We can't do that as part of codegen, because we can't know when the TSG is available nor the link to it.   With respect to the analyzer suggestion, I don't know how you would identify the pattern to make a generic analyzer out of it.

File: sdk/appconfiguration/Azure.Data.AppConfiguration/tests/ConfigurationLiveTests.cs (hunk 2 above)
Comment (jorgerangel-msft) [reply sdk/appconfiguration/Azure.Data.AppConfiguration/tests/ConfigurationLiveTests.cs]: I've removed the changes from the live test and only added a mock test.

**Review by kinelski — APPROVED** [review]
Changes LGTM!

**Review by jsquire — APPROVED** [review]
LGTM
